src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -204,6 +213,11 @@ def _run_binsearch_scaling(
                 high = new_size
                 midval = (high + low) // 2
                 new_size, _ = _adjust_batch_size(trainer, batch_arg_name, value=midval, desc="failed")
+
+                # Force the train dataloader to reset as the batch size has changed
```

Comment:
The reset is always preceded by `adjust_batch_size`. Should it be inside?

Reply:
not really. we call this while initialize the init_value of batch_size. I will improve this in a follow-up once batch-size finder callback PR is merged.